rtf_vis_tool/src/Components/frustum_classes/view_frustum.js

```diff
@@ -0,0 +1,168 @@
+/*Contains functions to transform camera instrinsic variables into a frustum consisting of 5 vertices.
+
+Author: Adi Singh
+*/
+
+// Third-Party Package Imports.
+var nj = require('numjs');
+
+class ViewFrustum {
+    // Generates vertices of a 5-face mesh for drawing a pinhole camera in 3d.
+
+    constructor(fx, img_w, img_h, ray_len) {
+        /*Initializes the ViewFrustum Object.
+
+            Args:
+                fx (float): Focal length in x-direction.
+                img_w (float): Image width (in pixels).
+                img_h (float): Image height (in pixels).
+                ray_len (float): Extent to which frustum rays extend away from optical center.
+        */
+
+        this.fx_ = fx;
+        this.img_w_ = img_w;
+        this.img_h_ = img_h;
+        this.frustum_ray_len_ = ray_len;
+    }
+
+    normalize_ray_dirs(ray_dirs) {
+        /*Normalizes the 5 ray directions of the frustum vertices to be of unit length.
+
+            Args:
+                ray_dirs: Array of shape (5,3) representing ray directions in camera frame.
+            Returns:
+                An array of shape (5,3) with normalized ray directions.
+        */
+
+        for (var row = 0; row < ray_dirs.shape[0]; row++) {
+            var vector_mag = Math.sqrt(Math.pow(ray_dirs.get(row,0),2) + Math.pow(ray_dirs.get(row,1),2) + Math.pow(ray_dirs.get(row,2),2));
```

Comment:
does numjs not have a `linalg.norm` fn?

Reply:
Nope, I couldn't find any documentation calculating magnitudes or norms of vectors.